### PR TITLE
taxa barplots

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

File: qp_qiime2/qiime2.py [hunk 1]
```diff
@@ -434,6 +434,98 @@ def alpha_correlation(qclient, job_id, parameters, out_dir):
...
         return False, None, error_msg
 
-    ainfo = [ArtifactInfo('o-visualization', 'q2_visualization',
+    ainfo = [ArtifactInfo('q2_visualization', 'q2_visualization',
                           [(o_visualization, 'qiime2-visualization')])]
     return True, ainfo, ""
+
+
+def taxa_barplot(qclient, job_id, parameters, out_dir):
+    """generate taxa barplot calculations
```
Comment (ElDeveloper) [inline qp_qiime2/qiime2.py]: generate -> Generate

File: qp_qiime2/qiime2.py [hunk 2]
```diff
@@ -434,6 +434,98 @@ def alpha_correlation(qclient, job_id, parameters, out_dir):
...
+        The path to the job's output directory
+
+    Returns
+    -------
+    boolean, list, str
+        The results of the job
+    """
+    out_dir = join(out_dir, 'taxa_barplot')
+    if not exists(out_dir):
+        mkdir(out_dir)
```
Comment (ElDeveloper) [inline qp_qiime2/qiime2.py:463]: Can we guarantee that all subdirectories will exist, or should this use `mkdirs` instead?

File: qp_qiime2/qiime2.py [hunk 3]
```diff
@@ -434,6 +434,98 @@ def alpha_correlation(qclient, job_id, parameters, out_dir):
...
+    # getting the biom table so we can check for taxonomies
+    biom_fp = artifact_info['files']['biom'][0]
+    bt = load_table(biom_fp)
+    with open(taxonomy_txt, 'w') as fp:
+        fp.write('Feature ID\tTaxon\n')
+        for otu_id in bt.ids('observation'):
+            tax = bt.metadata(id=otu_id, axis='observation')
+            if tax is None:
+                error_msg = ("biom table doesn't have taxonomy")
+                return False, None, error_msg
```
Comment (ElDeveloper) [inline qp_qiime2/qiime2.py:489]: What is going to happen with the taxonomy text file at this point?

**Review by ElDeveloper — CHANGES_REQUESTED** [review]
Can you remvoe the file `dump.rdb` or is this file needed for something?

File: qp_qiime2/qiime2.py (hunk 3 above)
Comment (antgonza) [reply qp_qiime2/qiime2.py:489]: Nothing, just an error raised.

File: qp_qiime2/qiime2.py (hunk 2 above)
Comment (antgonza) [reply qp_qiime2/qiime2.py:463]: Yes, it's guaranteed

File: qp_qiime2/qiime2.py (hunk 1 above)
Comment (antgonza) [reply qp_qiime2/qiime2.py]: k

**Pull request merged**